libnetwork/drivers/bridge/setup_ip_tables.go

```diff
@@ -414,6 +414,11 @@ func setupInternalNetworkRules(bridgeIface string, addr *net.IPNet, icc, insert
 	if err := programChainRule(version, outDropRule, "DROP OUTGOING", insert); err != nil {
 		return err
 	}
+	//Add internal interface to docker zone if ICC is enabled
+	if icc {
+		iiptable := iptables.GetIptable(version)
+		iiptable.SetupInternalICC(bridgeIface, insert)
+	}
 	// Set Inter Container Communication.
 	return setIcc(version, bridgeIface, icc, insert)
```

Comment:
Have yet to look at the code in context, but wondering if the changes should be part of this function. IIUC, the new changes would mean that things become less atomic; first some iptables changes are made, then we call `setIcc()`; probably those two steps should either "both fail" or "both success", and not "one of them succeed (or fail)"

Reply:
> become less atomic

I wouldn't worry too much about that. If any part of bridge network creation fails, [the driver throws out all its state about the network](https://github.com/moby/moby/blob/2cf63891427d68661bb8e3c58969058c62dbc85d/libnetwork/drivers/bridge/bridge.go#L701-L708) without invoking any of the [registered cleanup callbacks.](https://github.com/moby/moby/blob/2cf63891427d68661bb8e3c58969058c62dbc85d/libnetwork/drivers/bridge/bridge.go#L139) No aspect of bridge network creation is atomic.